Modules/_ctypes/callproc.c

```diff
@@ -1019,7 +1019,7 @@ void _ctypes_extend_error(PyObject *exc_class, const char *fmt, ...)
 
     PyErr_Fetch(&tp, &v, &tb);
     PyErr_NormalizeException(&tp, &v, &tb);
-    cls_str = PyObject_Str(tp);
+    cls_str = PyType_GetName((PyTypeObject *)tp);
```

Comment:
I'm not sure it's guaranteed that `tp` is actually a type here, although of course it should be. Perhaps we can execute this block only `if (PyType_Check(tb))`.

Reply:
See https://docs.python.org/3/c-api/exceptions.html?highlight=pyerr_fetch#c.PyErr_Fetch, tp is guaranteed to be a typeobject here.

Reply:
There's no explicit statement that says that, and there is no check in the C code in `errors.c` that ensures that the type is really a type.

Reply:
Fair enough. Corrected 👍